src/i_sound.c

```diff
@@ -474,6 +474,15 @@ void I_InitSound(void)
             from->volume = 0;
         }
     }
+
+    for (int i = sfx_sawup; i <= sfx_sawhit; i++)
+    {
+        S_sfx[i].pitch_range = PITCHRANGE_HALF;
+    }
+    S_sfx[sfx_itemup].pitch_range = PITCHRANGE_NONE;
+    S_sfx[sfx_tink].pitch_range = PITCHRANGE_NONE;
+    S_sfx[sfx_barexp].pitch_range = PITCHRANGE_NONE;
+    S_sfx[sfx_stnmov].pitch_range = PITCHRANGE_NONE;
```

Comment:
Is it possible for someone to use dehacked or similar to repurpose these sounds for monsters? Would that cause players to think the variable pitch feature isn't working correctly? I ask because of something similar that happened with 3d audio: [doomworld](https://www.doomworld.com/forum/post/2744652)

Reply:
Yes, sure, if we go by sfx number, then this may happen. We could either recommend to turn pitch emulation off for such mods, or try a more sophisticated context based approach.

Reply:
I think only the explosion sound needs this kind of consideration; the platform move sound is fixed so I think it's unlikely to be repurposed for any customized actors.
Maybe even since icon of sin dying is comparatively much rarer than a multisector platform moving we could just let that one go.

Reply:
I think we can do that, i.e. disable pitch shifting for the explosion sound if it's emitted as part of the icon of sin explosion. 

How does the situation at the start of Doom 2 MAP23 sound like to you, when the dozens of barrels explode in short distance?

Reply:
It isn't really an issue there, I think probably because of the random duration first state when the barrels die before they explode

Reply:
Right. So, one thing we could try is `S_sfx[sfx_barexp].pitch_range = PITCHRANGE_HALF` to reduce the amount of pitch shifting or somehow set a global variable (or similar approach) to tell `P_ExplodeMissile()` to disable pitch shifting for `mo->info->deathsound`.

Reply:
Should be fixed now.